qdrant-landing/content/articles/memory-consumption.md

```diff
@@ -0,0 +1,366 @@
+```
+
+## Let's run some benchmarks
+
+Let's run some benchmarks to see how much RAM Qdrant needs to serve 1 million vectors.
+
+We can use the `glove-100-angular` and scripts from the [vector-db-benchmark](https://github.com/qdrant/vector-db-benchmark) project to upload and query the vectors.
+With the first run we will use the default configuration of Qdrant with all data stored in RAM.
+
+```bash
+# Upload vectors
+python run.py --engines qdrant-all-in-ram --datasets glove-100-angular
+```
+
+After uploading vectors, we will repeat the same experiment with different allowed RAM usage to see how they affect the memory consumption and search speed.
+
+```bash
+# Search vectors
+python run.py --engines qdrant-all-in-ram --datasets glove-100-angular --skip-upload
+```
+
+<!--
+
+Experiment results:
+
+All in memory:
+
+1024mb - out of memory
+1512mb - 774.38 rps
+1256mb - 760.63 rps 
+1152mb - out of memory
+1200mb - 794.72it/s
+
+Conclusion: about 1.2Gb is needed to serve ~1 million vectors, no speed degradation with limiting memory above 1.2Gb
+
+MMAP for vectors:
+
+1200mb - 759.94 rps
+1100mb - 687.00 rps
+1000mb - 10 rps
+
+--- use a bit faster disk ---
+
+1000mb - 25 rps
+500mb - out of memory
+750mb - 5 rps
+625mb - 2.5 rps
+575mb - out of memory
+600mb - out of memory
+
+We can go even lower by using mmap not only for vectors, but also for the index.
+MMAP for vectors and HNSW graph:
+
+600mb - 5 rps
+300mb - 0.9 rps / 1.1 sec per query
+150mb - 0.4 rps / 2.5 sec per query
+75mb - out of memory
+110mb - out of memory
+125mb - out of memory
+135mb - 0.33 rps / 3 sec per query
+
+-->
+
+### All in Memory
+
+In the first experiment, we tested how well our system performs when all vectors are stored in memory.
+We tried using different amounts of memory, ranging from 1512mb to 1024mb, and measured the number of requests per second (rps) that our system was able to handle.
+
+| Memory | Requests/s    |
+|--------|---------------|
+| 1512mb | 774.38        |
+| 1256mb | 760.63        |
+| 1200mb | 794.72        |
+| 1152mb | out of memory |
+| 1024mb | out of memory |
+
+
+We found that 1152Mb memory limit resulted in our system running out of memory, but using 1512mb, 1256mb, and 1200mb of memory resulted in our system being able to handle around 780 RPS.
+This suggests that about 1.2Gb of memory is needed to serve around 1 million vectors, and there is no speed degradation when limiting memory usage above 1.2Gb.
+
+### Vectors stored using MMAP
+
+Let's go a bit further!
+In the second experiment, we tested how well our system performs when **vectors are stored using the memory-mapped file** (mmap).
+
+```http
+PUT /collections/benchmark 
+
+{
+  ...
+  "optimizers_config": {
+    "mmap_threshold_kb": 20000
+  }
+}
+
+```
+This configuration tells Qdrant to use mmap for vectors if the segment size is greater than 20000Kb (which is approximately 40K 128d-vectors).
+
+Now the out-of-memory happens when we use allow only **600mb** RAM.
```

Comment:
Shouldn't it be "Now the out-of-memory happens when we allow using **600mb** RAM only"?